GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,108 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+import android.text.TextUtils;
+
+public abstract class Preconditions {
+    
+    /**
+     * The goal of this class is to improve the readability and to provide some common checking method to improve 
+     * the robustness of GVRF. But in some cases, this may be a significant performance cost. Try to use and further
+     * extend this class carefully with concern on performance issue.
+     * 
+     * Actually, one purpose of this class may help us avoid from using GVRJniException which is not a proper way to 
+     * throw/simulate an exception with JNI crash.
+     */
+    
+    /** General parameter name for key String in checkStringNotNullOrEmpty */
+    public static final String INPUT_TYPE_KEY = "key";
+
+
+    /**
+     * Check that the if the parameter is {@code null}.
+     *  
+     * @param object
+     *          the parameter object can be any Java reference type, including arrays.
+     * @param parameterName
+     *          To point out the type of data, for the use when we try to throw an exception.
+     *          Type may not only refer to a class name but a wider concept of reference type such as vertices, normals.
+     * @throws IllegalArgumentException if the object is null pointer.
+     */
+    public static void checkNotNull(String parameterName, Object object) {
+        if (object == null) {
+            throw Exceptions.IllegalArgument("Input %s can't be null.",
+                    parameterName);
+        }
+    }
+    
+    /**
+     * This function is to ensure that any input string value is not null or empty
+     * @param value
+     *          String value to be checked.
+     * @param parameterName
+     *          The name/type of the string value.
+     *          
+     * @throws IllegalArgumentException if the key is null or empty.
+     */
+    public static void checkStringNotNullOrEmpty(String parameterName, String value){
+        if(value == null || TextUtils.isEmpty(value)){
```

Comment:
I see. Just updated the format. Don't know why but "Shift+Ctrl+F" doesn't work this time. I can only format this file by doing "Source -> Format"